### ISSUE TITLE
`treeInfo()`might be wrong if `class.values` are not ordered in ascending order

### ISSUE THREAD

**Issue by LeaKronziel** [imbs-hl/ranger#651]
It might be that something is wrong with the `treeInfo()` function. For classification trees, the given prediction of the terminal nodes changes when the order of observations of the Dependent Variable changes. This allows, for example, different predictions to be displayed when the `treeInfo()` function is used for visualization.

As an example, I created a dataset consisting of two **identical variables** `number1` and `number2` (just because I can't use ranger with only a single variable) that can take values from 1 to 100. For values up to 33 the dependent variable corresponds to the label `small`, for values above 63 to the value `large` and in between to the value `medium`. 
When the data is entered into `ranger()` in alphabetical ascending order (`large`, `medium`, `small`), the `class.values` correspond to values 1, 2, 3 and a plausible split is made in the `treeInfo()`. 

Here are the `class.values` and `levels`
![example_classvalues_asc](https://user-images.githubusercontent.com/115787989/212432763-290a8f67-5ca4-4109-9f6f-4cd31aa87fb6.png)

the `treeInfo`
<img width="655" alt="example_tree_info_asc" src="https://user-images.githubusercontent.com/115787989/212432781-079fdfbe-f041-45f5-a139-6a9eba534191.png">

and a visualisation of the `treeInfo` 
<img width="978" alt="tree_asc" src="https://user-images.githubusercontent.com/115787989/212432977-ab31b239-889a-4a05-ae58-c8783300e19d.png">

However, if the same data is sorted differently, for example descending, the `class.values` and the prediction in `treeInfo()` change. In this case, values below 33 are classified as `large`, which does not fit the training data. In this case, the `predict()` function gives the more reasonable output `small`.

Here are the `class.values` and `levels`
![example_classvalues_desc](https://user-images.githubusercontent.com/115787989/212433160-31c32a3e-c921-47a9-9041-e6395b17259e.png)

the `treeInfo`
<img width="655" alt="example_tree_info_desc" src="https://user-images.githubusercontent.com/115787989/212433118-d7d7adf1-ac37-4489-a739-d5819341c77d.png">

and a visualisation of the `treeInfo`
<img width="978" alt="tree_desc" src="https://user-images.githubusercontent.com/115787989/212433176-4084837b-f579-4a62-881a-6af5792bc9e7.png">

Obviously the second `treeInfo` gives wrong information in this case, although the same data was used.

Below is a minimal working example to reproduce the issue:
```R
library(dplyr)
library(ranger)

set.seed(42)
example_data <- data.frame(number1 = sample(1:100, 30, replace = TRUE)) %>%
  mutate(number2 = number1) %>%
  mutate(annotation = case_when(number1 <= 33 ~ "small",
                                 number1 > 33 & number1 <= 63 ~ "medium",
                                 number1 > 63 ~ "large")) %>%
  mutate(annotation = factor(annotation))

# Random forest with alphabetical ascending order
example_data_asc <- example_data %>% arrange(annotation)

example_rf_asc <- ranger(data = example_data_asc,
                         annotation ~.,
                         seed = 42)
example_tree_info_asc <- treeInfo(example_rf_asc)


# Random forest with alphabetical descending order
example_data_desc <- example_data %>% arrange(desc(annotation))

example_rf_desc <- ranger(data = example_data_desc,
                          annotation ~.,
                          seed = 42)

example_tree_info_desc <- treeInfo(example_rf_desc)

```

**Comment by mnwright** [comment]
Thanks. I think this was already fixed in #629. It's the same bug as in #626, right? 

However, the fix is not on CRAN yet. You'll have to install from Github for now, e.g. with `devtools::install_github("imbs-hl/ranger")`.

**Comment by LeaKronziel** [comment]
Sorry for the late reply. Yes, it's the same issue as #626. I tried the github version, which worked as expected.  Thank you.
